README.md

```diff
@@ -0,0 +1,193 @@
+optional parameters are expressible through the CLI.  Help is also handled by the cliapi framework.
+
+- **scoops**: a dictionary which maps a CLI query name to a particular API data scoop. "scoops" are really just
+"_sandboxed_ python _eval()_" statements.  This allows predefined scoops to be expressed as CLI options.
+It also allows for restricted ad-hoc queries to be provided on the command line when a return value is not
+currently supported as an option in the CLI.  The query uses Python's dictionary lookup syntax for slice
+slice**s** (See examples using **--query=** below).
+Python's _eval()_ function allows limiting access to a single data structure and "no builtins" through
+this facility:
+
+- **fetchers**: a dictionary which maps from a particular API name to the actual python function which
+provides the _backing store_ for the contents of the top-level API dictionary.
+
+### cliapi directory layout:
+```
+├── cliapi                      root of cliapi package
+│   ├── __init__.py
+│   ├── cliapi_lib.py           framework meta classes and decorators
+│   ├── cliapi.py               main() and the CLI generation code
+│   └── what_cloud.py           module (useful for detecting which cloud plugins are valid)
+│   ├── providers               directory for plugin providers
+│   │   ├── __init__.py
+│   │   ├── azure.py            plugin for Azure/SUSE APIs
+│   │   ├── <your plugin here>  ...your plugin goes here
+│   │   ├── ...                 ...additional plugins are automatically discovered by cliapi
+│   │   └── test.py             ..."because it's not a framework without at least 2 plugins"
+├── design.md                   design considerations for cliapi
+├── LICENSE                     provisional license (Apache2 is mutable into any other license)
+├── README.md                   this document
+└── setup.py                    Python installation script
+```
+
+### Examples:
+
+**example #1** - Common help AND plugin provider help (default provider = azure)
+```
+ed-sle12sp3byos:/home/lane/cliapi # cliapi --help
+usage: /usr/bin/cliapi [display option#1]... [API option#1]... [CLI option]
+
+***[ azure ]*** provider Display options:
+  --internal-ip                     
+  --location         region location
+  --cloud-service    what           
+  --instance-name    name of instance
+  --mac              the MAC address for this interface
+  --external-ip                     
+
+***[ azure ]*** provider API config options:
+  --api_version=     default='2017-08-01', azure metadata api version
+
+Common CLI options:
+  --help             help for this CLI command
+  --provider=        specify name of provider module
+  --list-providers   list all available providers
+  --list-apis        list all available APIs for specified provider
+  --query=           specify a python dictionary style query command
+  --all              output all API results for specified API options or defaults
```

Comment:
'--all' should provide the information for one API version, not all information for all API versions. This may just be a wording issue. However, given the framework itself is plugable I am not certain that all these options would apply to all plugins. A more flxible approach might be to have a get_general_options() method in the plugins that allows each plugin to specify certain options that should be shown as general options rather than plugin specific options, although they are of course plugin specific.